### PR TITLE
Media: Add support for hotlinking audio files

### LINKED ISSUE #10888: Media: Add support for hotlinking audio files
## Feature Description

Breakout ticket from #10515. Allow users select hotlink audio files served from a different source for background audio functionality. Like video hotlinking, user may want to offload audio assets to third party service to host large audio files. Unlike video, audio is not stored in google cache either.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Allow users select hotlink audio files served from a different source for background audio functionality. 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Enable audio hotlinking experiment. 
2. Create new story. 
3. Select page. 
4. Go style tab. 
5. Click link Link to audio file. 
6. Input URL - https://ia800905.us.archive.org/25/items/EDIS-SRP-0197-06/EDIS-SRP-0197-06.mp3.
7. Click Use audio file button. 
8. Play / pause audio. Hear audio. 

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10888

### CODE REVIEW

**Comment by timarney** [comment]
I see this is still in draft but ... also see 

> [spacedmonkey ](https://github.com/spacedmonkey)requested your review on this pull request.

So gave this a quick lookover + test

Using https://samplelib.com/lib/preview/mp3/sample-6s.mp3 No CORS error was thrown  in the UI --- I see a play, pause button + errors in the console

Screen Shot 2022-04-11 at 5 02 29 AM

File: packages/story-editor/src/components/panels/shared/hotlinkModal.js [hunk 1]
```diff
@@ -134,7 +135,12 @@ function HotlinkModal({
         needs_proxy: shouldProxy,
       });
 
-      onSelect({ src: link, needsProxy: shouldProxy });
+      onSelect({
+        ...hotlinkInfo,
+        id: uuidv4(),
```
Comment (swissspidy) [inline packages/story-editor/src/components/panels/shared/hotlinkModal.js]: What's the ID for? 🤔 
I don't think we should start adding IDs to hotlinked media items unless there is a valid use case.

Comment (spacedmonkey) [reply packages/story-editor/src/components/panels/shared/hotlinkModal.js]: Removed

Comment (spacedmonkey) [reply packages/story-editor/src/components/panels/shared/hotlinkModal.js]: FIxed.

**Comment by spacedmonkey** [comment]
@timarney I have reviewed this PR and it is ready for review again.

**Comment by timarney** [comment]
Seems to be working well now 👍  

PR desc needs test instructions etc... but code looks good.

**Comment by swissspidy** [comment]
Successfully tested with a file like https://ia800905.us.archive.org/25/items/EDIS-SRP-0197-06/EDIS-SRP-0197-06.mp3.

**Related:**

Just tried this with the audio file https://storage.googleapis.com/media-session/sintel/snow-fight.mp3 and it didn't work because it has the mime type `audio/mp3` and not the expected `audio/mpeg`.

`audio/mp3` is not in the `wp_get_mime_types()` allowlist, so that makes sense.

If this turns out to be something our users run into frequently we might need to add some workaround for it.

**Review by swissspidy — DISMISSED** [review]

**Comment by timarney** [comment]
Going to approve and just noting (as it was mixed into the other comment):

> PR desc needs test instructions etc... but code looks good.

https://github.com/GoogleForCreators/web-stories-wp/pull/11221#issuecomment-1105848971

**Review by timarney — DISMISSED** [review]

**Comment by spacedmonkey** [comment]
> PR desc needs test instructions

Testing instructions added.

**Review by swissspidy — DISMISSED** [review]

**Review by swissspidy — APPROVED** [review]

**Pull request merged**